include/engine/routing_algorithms/routing_base_ch.hpp

```diff
@@ -350,63 +350,29 @@ void retrievePackedPathFromSingleHeap(const SearchEngineData<Algorithm>::QueryHe
 // && source_phantom.GetForwardWeightPlusOffset() > target_phantom.GetForwardWeightPlusOffset())
 // requires
 // a force loop, if the heaps have been initialized with positive offsets.
-void search(const datafacade::ContiguousInternalMemoryDataFacade<Algorithm> &facade,
+void search(SearchEngineData<Algorithm> &engine_working_data,
```

Comment:
Passing `engine_working_data` as another parameter complicates the interface more, and makes it even less unit test-table. However I see why we need this right now.

Ideally I would like to see something like:
```cpp
std::pair<EdgeWeight, std::vector<NodeID>>
search(GraphT& graph,
       HeapT&,
       HeapT&,
       std::array<PhantomNode, N1>& sources,
       std::array<PhantoNode, N2>& targets);
```
Which would enable us to write unit tests for these functions (no `facade` and `engine_working_data` dependency). We should aim for this in `5.8`.